### PR TITLE
Add missing tests for meditrak-app-server

### PR DESCRIPTION
**Pull request by rohan-bes**
Just a couple of routes that missed having tests added amidst all the PRs. Also fixed a couple small bugs in the routes which is nice 😄

### CODE REVIEW

File: packages/database/src/testUtilities/buildAndInsertSurveyResponses.js [hunk 1]
```diff
@@ -67,9 +67,11 @@ const buildAndInsertSurveyResponse = async (
...
-      buildAndInsertSurveyResponse(models, user, surveyResponse),
-    ),
-  );
+  const builtResponses = [];
+  for (let i = 0; i < surveyResponses.length; i++) {
+    const surveyResponse = surveyResponses[i];
+    const builtResponse = await buildAndInsertSurveyResponse(models, user, surveyResponse);
+    builtResponses.push(builtResponse);
+  }
+  return builtResponses;
```
Comment (rohan-bes) [inline packages/database/src/testUtilities/buildAndInsertSurveyResponses.js:76]: This caused what must be a rare bug on my machine: 
`error: duplicate key value violates unique constraint "entity_code_key"`

I _think_ it's due to `Promise.all` executing operations in non-deterministic order, which might've lead to a situation where two `findOrCreate()` calls were made at the same time. I guess the real underlying issue here is that `DatabaseModel.findOrCreate()` is non-atomic... but that seems like a bit too big of an issue to tackle as part of this PR...

File: packages/meditrak-app-server/src/app/createApp.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ export function createApp(database = new TupaiaDatabase()) {
     .post<RegisterUserRequest>('user', handleWith(RegisterUserRoute))
     .get<SocialFeedRequest>('socialFeed', authMiddleware, handleWith(SocialFeedRoute))
     .get<UserRewardsRequest>('me/rewards', authMiddleware, handleWith(UserRewardsRoute))
-    .get<ChangePasswordRequest>(
+    .post<ChangePasswordRequest>(
```
Comment (rohan-bes) [inline packages/meditrak-app-server/src/app/createApp.ts:32]: Whoops! Should've been `post` all along!

File: packages/meditrak-app-server/src/routes/UserRewardsRoute.ts [hunk 3]
```diff
@@ -22,8 +22,10 @@ export class UserRewardsRoute extends Route<UserRewardsRequest> {
     const [{ coconuts, pigs }] = await models.database.executeSql<
       [{ coconuts: number; pigs: number }]
     >(
+      // The COUNT(*)::int is required here, since pg serializes count to string
+      // (https://stackoverflow.com/questions/47843370/postgres-sequelize-raw-query-to-get-count-returns-string-value)
```
Comment (rohan-bes) [inline packages/meditrak-app-server/src/routes/UserRewardsRoute.ts:26]: Funny weird bug here!

File: packages/database/src/testUtilities/buildAndInsertSurveyResponses.js (hunk 1 above)
Comment (IgorNadj) [reply packages/database/src/testUtilities/buildAndInsertSurveyResponses.js:76]: Ah that's no good! Can you create a card for it?

**Review by IgorNadj — APPROVED** [review]
Nice one

File: packages/database/src/testUtilities/buildAndInsertSurveyResponses.js (hunk 1 above)
Comment (rohan-bes) [reply packages/database/src/testUtilities/buildAndInsertSurveyResponses.js:76]: Good idea! Have created and added to refinement: https://linear.app/bes/issue/RN-536/rare-bug-where-creating-entities-fail-when-many-attempts-are-made-at

**Pull request merged**